src/sentry/static/sentry/app/components/stackedBarChart.jsx

```diff
@@ -118,31 +82,64 @@ const StackedBarChart = createReactClass({
           throw new Error('Only one of [points|series] should be specified.');
         }
 
-        series = StackedBarChart.pointsToSeries(nextProps.points);
+        series = this.pointsToSeries(nextProps.points);
       }
 
       this.setState({
         series,
-        pointIndex: StackedBarChart.pointIndex(series),
-        interval: StackedBarChart.getInterval(series),
+        pointIndex: this.pointIndex(series),
+        interval: this.getInterval(series),
       });
     }
-  },
+  }
 
   shouldComponentUpdate(nextProps, nextState) {
     return !_.isEqual(this.props, nextProps);
```

Comment:
is this really necessary? seems like a) this would be close to the default behavior and b) it would be a confusing surprise to not re-render on state change.

Reply:
I'm genuinely asking tho (hence why I left it in)

Reply:
Difference is shallow vs deep compare, agreed with (b).

Since we're not updating on state changes... the values that are currently in state, should not be in state. I would leave this in here though and open a followup for cleanups.